src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -569,8 +569,18 @@ ValidatorList::for_each_listed (
 }
 
 std::size_t
-ValidatorList::calculateQuorum (std::size_t nTrustedKeys)
+ValidatorList::calculateQuorum (
+    std::size_t nTrustedKeys, std::size_t nSeenValidators)
 {
+    // Check that lists from all configured publishers are available
```

Comment:
This comment and the comment inside the `for` should be combined. I'd replace this with the comment from inside the `for`.

Also, I wondered whether this looked cleaner:

```
    // Do not use achievable quorum unless lists from all configured
    // publishers are available.
    if (! std::all_of (publisherLists_.begin(), publisherLists_.end(),
            [](auto const& e)
            {
                return e.second.available;
            }))
        return std::numeric_limits<std::size_t>::max();
```

I don't think it does.